pkg/deploy.go

```diff
@@ -0,0 +1,189 @@
+/*
+SPDX-License-Identifier: Apache-2.0
+
+Copyright Contributors to the Submariner project.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package pkg
```

Comment:
Same here re: this directly under `pkg`.  It looks like the code here deploys the broker so I'd suggest _pkg/broker_.

Reply:
Having it under `pkg/broker` causes cyclic imports with `pkg/discovery`. I am thinking of putting `deploy.go` and `join.go` into a separate package. Suggestions for its name are welcome :)

Reply:
We could have a `deploy` package with functions `Broker` etc. Or we try to resolve the cyclic import re: `pkg/broker`.